frontend/packages/knative-plugin/src/components/add/event-sources/KafkaSourceSection.tsx

```diff
@@ -1,37 +1,121 @@
 import * as React from 'react';
-import { useFormikContext, FormikValues } from 'formik';
+import * as _ from 'lodash';
 import FormSection from '@console/dev-console/src/components/import/section/FormSection';
-import { InputField, TextColumnField } from '@console/shared';
+import { InputField, SelectInputField, SelectInputOption } from '@console/shared';
 import { TextInputTypes } from '@patternfly/react-core';
 import KafkaSourceNetSection from './KafkaSourceNetSection';
 import ServiceAccountDropdown from '../../dropdowns/ServiceAccountDropdown';
+import { KafkaModel, KafkaTopicModel } from '../../../models';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { getBootstrapServers } from '../../../utils/create-eventsources-utils';
 
 const KafkaSourceSection: React.FC = () => {
-  const { values } = useFormikContext<FormikValues>();
-  const {
-    data: {
-      kafkasource: { bootstrapServers, topics },
-    },
-  } = values;
+  const [bootstrapServers, setBootstrapServers] = React.useState<SelectInputOption[]>([]);
+  const [bsPlaceholder, setBsPlaceholder] = React.useState<React.ReactNode>('');
+  const [kafkaTopics, setKafkaTopics] = React.useState<SelectInputOption[]>([]);
+  const [ktPlaceholder, setKtPlaceholder] = React.useState<React.ReactNode>('');
+
+  const kafkaResource = React.useMemo(
+    () => ({
+      isList: true,
+      kind: referenceForModel(KafkaModel),
+      optional: true,
+    }),
+    [],
+  );
+  const kafkaTopicResource = React.useMemo(
+    () => ({
+      isList: true,
+      kind: referenceForModel(KafkaTopicModel),
+      optional: true,
+    }),
+    [],
+  );
+  const [kafkasData, kafkasLoaded, kafkasLoadError] = useK8sWatchResource<K8sResourceKind[]>(
+    kafkaResource,
+  );
+  const [kafkaTopicsData, kafkaTopicsLoaded, kafkaTopicsLoadError] = useK8sWatchResource<
+    K8sResourceKind[]
+  >(kafkaTopicResource);
```

Comment:
why not use `useK8sWatchResources` here? it can take an array of resources.

Reply:
Great, will use that then. Didnt know we had that too :P